core/src/avm1/activation.rs

```diff
@@ -316,4 +337,34 @@ impl<'gc> Activation<'gc> {
             }
         }
     }
+
+    /// Attempts to lock the activation frame for execution.
+    ///
+    /// If this frame is already executing, that is an error condition.
+    pub fn lock(&mut self) -> Result<(), Error> {
+        if self.is_executing {
+            return Err("Attempted to execute the same frame twice".into());
+        }
+
+        self.is_executing = true;
+
+        Ok(())
+    }
+
+    /// Unlock the activation object. This allows future execution to run on it
+    /// again.
+    pub fn unlock_execution(&mut self) {
+        self.is_executing = false;
+    }
```

Comment:
Bikeshed: `lock` and `unlock` or  `lock_execution` and `unlock_execution` would be more symmetrical.